docs/reference/plugin-api.md

```diff
@@ -196,11 +196,15 @@ In client component:
 
 ```js
 import { usePageData } from '@vuepress/client'
+import { useResolveRouteWithRedirect } from '@vuepress/theme-default'
 
 export default {
   setup() {
     const page = usePageData()
-    console.log(page.value.meta) // foobar
+    const route = useResolveRouteWithRedirect('/')
+
+    console.log(page.value.meta.foo) // 'bar'
+    console.log(route.meta.path) // '/'
```

Comment:
As `meta` is already available in the `route.meta`, is it necessary to put `meta` in to page data? 

I think we can make `meta` a property of the `Page` object, and don't need to put it into page data.

What's your opinion? @Mister-Hope

Reply:
Sounds ok to me.

Reply:
If we remove it from page data, we could not use `extendsPageData` to modify it. I have two proposals:

1. Let users to modify it manually in `onInitialized` hook:

```ts
onInitialized: (app) => app.pages.forEach(....)
```

In fact, the `extendsPageData` hook could also be removed as we have added `page.data` property, and users can modify it in this way, too. 😅 

2. Rename `extendsPageData` to `extendsPage`:

```ts
extendsPage: (page) => {
  page.data.foo = 'foo'
  page.routeMeta.title = page.title
  // ...
}
```

The only difference is that plugins / themes don't need to traverse the `app.pages` array by themselves.

---

@Mister-Hope